packages/core/src/odata/common/filter/number-filter-function.ts

```diff
@@ -22,10 +22,9 @@ export class NumberFilterFunction<
   constructor(
     functionName: string,
     parameters: FilterFunctionParameterType<EntityT>[],
-    edmType: 'Edm.Int32' | 'Edm.Double' | 'Edm.Decimal'
+    public edmType: 'Edm.Int32' | 'Edm.Double' | 'Edm.Decimal'
```

Comment:
Is see that you need the type for example in the `convertToUriFormat(filter.value, filter.edmType!)`. However, could you either make it `readonly` or use a getter function to ensure that you do not fiddle around with the type. Consider also to make the other parameter:
```
 constructor(
    public field: string | FilterFunction<EntityT, FieldT>,
    public operator: FilterOperator,
    public value: FieldT,
    public edmType?: EdmTypeShared<ODataVersionOf<EntityT>>
  ) {
```
readonly.

Reply:
done.